### ISSUE TITLE
how to use gmm in other object detection model?

### ISSUE THREAD

**Issue by Shanyaodedanshen** [NVlabs/AL-MDN#2]
Thx for ur work.
How to use gmm in other object detection model? Such as RetinaNet, ATSS, GFLv2.

**Comment by jwchoi384** [comment]
Hello,
I haven't tried our proposed modeling for the algorithm you mentioned. 
However, I think that the proposed method can be applied to other networks because most of the output of object detection algorithm is the same.
It worked well when I applied it to Faster-RCNN.

- First, increase the number of output size of both classification and localization heads by considering the number of Gaussian mixture model (GMM) components.
- Second, modify the loss function as the proposed loss function with the GMM parameters of each head.
- Lastly, after training is completed, modify the calculation of bounding box coordinates and classification information considering GMM in inference.

**Comment by Shanyaodedanshen** [comment]
> Hello, I haven't tried our proposed modeling for the algorithm you mentioned. However, I think that the proposed method can be applied to other networks because most of the output of object detection algorithm is the same. It worked well when I applied it to Faster-RCNN.
> 
> * First, increase the number of output size of both classification and localization heads by considering the number of Gaussian mixture model (GMM) components.
> * Second, modify the loss function as the proposed loss function with the GMM parameters of each head.
> * Lastly, after training is completed, modify the calculation of bounding box coordinates and classification information considering GMM in inference.

Thank you. I have apply your work in RetinaNet.
But When I use the raw SSD+GMM for BDD dataset, mAP with only 2.+, which use the same parameters with COCO.
And when I use RetinaNet+GMM for BDD dataset, mAP with only 8.+. However, the common mAP for BDD dataset is about 20+. Could u help me...

**Comment by Shanyaodedanshen** [comment]
> > Hello, I haven't tried our proposed modeling for the algorithm you mentioned. However, I think that the proposed method can be applied to other networks because most of the output of object detection algorithm is the same. It worked well when I applied it to Faster-RCNN.
> > 
> > * First, increase the number of output size of both classification and localization heads by considering the number of Gaussian mixture model (GMM) components.
> > * Second, modify the loss function as the proposed loss function with the GMM parameters of each head.
> > * Lastly, after training is completed, modify the calculation of bounding box coordinates and classification information considering GMM in inference.
> 
> Thank you. I have apply your work in RetinaNet. But When I use the raw SSD+GMM for BDD dataset, mAP with only 2.+, which use the same parameters with COCO. And when I use RetinaNet+GMM for BDD dataset, mAP with only 8.+. However, the common mAP for BDD dataset is about 20+. Could u help me...

I use NLL loss and smooth l1 loss meanwhile. Firstly give smooth l1 loss more weight. After the location feature has been trained well, give NLL loss more weight.
I also use focal loss, use logits=mu+var*random as the prediction input of focal loss.
the result has been normal.

**Comment by abhigoku10** [comment]
@Shanyaodedanshen  hi i am looking into similar work can you pls share your work ?
